### DISCUSSION TITLE
Update single file using RAUC

### DISCUSSION THREAD

**Discussion by AdityaICVS** [rauc/rauc#1403] [Q&A]
Hello,

From the post #497 and #687, I got that RAUC is not designed to implement to single file updates. I can probably do something with pre-install hook functionality, but I haven't tried that and not sure how the update will go.

We are designing one system where we want to have provision for having full system updates (Kernel, RootFS) and partial updates or single file updates (binaries, jpeg, xml). 

I want to update the device using USB drive/pendrive. It is possible to achieve using RAUC using CAsync? 

Can we have a system/partition configuration as following?
1. Kernel0
2. Rootfs0
3. Update0
4. Appfs0
5. Appfs1

where RAUC performs updates from **Update0** partition to **Kernel0**, **Rootfs0**, **Appfs0**and **Appfs1** with parent as **Rootfs0**?

Thanks.

**Comment by enienws** [comment]
I also try to achieve single file update with RAUC. The parameter allow-mounted=<true/false> for slots defined in the documentation [https://rauc.readthedocs.io/en/latest/reference.html#slot-slot-class-idx-sections](url) makes me to think updating single files is possible with RAUC. For slots where allow-mounted=true and with a custom hook I think It can be possible to update single files within a mount point. 
I haven't tried it yet but any confirmation will be great before dig into the implementation details.
